### PR TITLE
STORM-2361 Kafka spout - after leader change, it stops committing offsets to ZK

### PR DESCRIPTION
**Pull request by ernisv**
Try to route acks/fails to recreated partition managers if manager is not found by exact (broker, partition, topic) combination

### CODE REVIEW

**Comment by ptgoetz** [comment]
@ernisv Can you update the code to put all for, if-else, etc. blocks curly braces?

**Comment by ernisv** [comment]
Sure - braces added.

File: external/storm-kafka/src/jvm/org/apache/storm/kafka/KafkaSpout.java [hunk 1]
```diff
@@ -172,6 +187,12 @@ public void fail(Object msgId) {
         PartitionManager m = _coordinator.getManager(id.partition);
         if (m != null) {
             m.fail(id.offset);
+        } {
```
Comment (HeartSaVioR) [inline external/storm-kafka/src/jvm/org/apache/storm/kafka/KafkaSpout.java]: missing `else` between `} {`

Comment (ernisv) [inline external/storm-kafka/src/jvm/org/apache/storm/kafka/KafkaSpout.java]: Right - fixed.

**Comment by HeartSaVioR** [comment]
+1 
@ernisv Could you squash commits into one? 
Btw, I feel this patch should be along with STORM-2296, so if we would want to port this back to 1.0.x, STORM-2296 should be ported back first.

**Comment by ernisv** [comment]
Squashed the commits.
About moving to STORM-2296 - I've put the links/relations to STORM-2296.
But as the issue itself is already in "resolved" state - not sure how the patch could be added to it.
Is registering the relation not enough?

**Comment by HeartSaVioR** [comment]
@ernisv 
Could you change your commit title to be meaningful? We often set commit title to same as JIRA title.
I can take care of other things so you don't need to worry about that.

**Comment by ernisv** [comment]
Ok, changed the title

**Pull request merged**